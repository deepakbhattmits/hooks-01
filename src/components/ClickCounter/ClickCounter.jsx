import React, { useState } from 'react'
// import Counter from '../UI/Counter'
const ClickCounter = () => {
    const [counter, setCounter] = useState(0);
    const handleClick = () => {
        setCounter(counter + 1)
    }
    return (
        <div data-test="component-click-counter">
            <h1 dataTest='counter-display'>
                {counter}
            </h1>
            <button data-test="increment-button" type="submit" onClick={handleClick}>
                increment button
            </button>
        </div>
    )
}
export default ClickCounter
