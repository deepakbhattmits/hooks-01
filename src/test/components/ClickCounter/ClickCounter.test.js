import React from 'react'
import { shallow } from "enzyme";
import ClickCounter from '../../../components/ClickCounter/ClickCounter';
// import Button from '../../../components/UI/Button';

/**
 * Factory function to create a ShallowWrapper for the App component.
 * @function setup
 * @param {object} props - Component props specific to this setup.
 * @param {object} state - Initial state for setup.
 * @returns {ShallowWrapper}
*/
const setup = (props = {}, state = null) => {
    const wrapper = shallow(<ClickCounter {...props} />)
    if (state) wrapper.setState(state);
    return wrapper;
}

/**
 * Return ShallowWrapper containing node(s) with the given data-test value.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within.
 * @param {string} val - Value of data-test attribute for search.
 * @returns {ShallowWrapper}
 */
const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
}


test('renders without crashing', () => {
    const wrapper = setup();
    const clickCountWrapper = findByTestAttr(wrapper, 'component-click-counter');
    expect(wrapper.debug()).toMatchSnapshot();
    expect(clickCountWrapper.length).toBe(1);
});
test('renders with increment button', () => {
    const wrapper = setup();
    const clickCountWrapper = findByTestAttr(wrapper, 'increment-button');
    expect(wrapper.debug()).toMatchSnapshot();
    expect(clickCountWrapper.length).toBe(1);
});
test("renders counter initial value", () => {
    const wrapper = setup();
    // const initialStateCounter = wrapper.text();
    const initialStateCounter = findByTestAttr(wrapper, 'counter-display');
    console.log('TEST : ', initialStateCounter)
    expect(initialStateCounter.textContent).toBe('0')
    // console.log('TEST : ', initialStateCounter)
    // expect(initialStateCounter).toBe({});
});

