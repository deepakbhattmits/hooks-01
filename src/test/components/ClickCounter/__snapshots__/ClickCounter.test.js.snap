// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with increment button 1`] = `
"<div data-test=\\"component-click-counter\\">
  <h1 dataTest=\\"counter-display\\">
    0
  </h1>
  <button data-test=\\"increment-button\\" type=\\"submit\\" onClick={[Function: handleClick]}>
    increment button
  </button>
</div>"
`;

exports[`renders without crashing 1`] = `
"<div data-test=\\"component-click-counter\\">
  <h1 dataTest=\\"counter-display\\">
    0
  </h1>
  <button data-test=\\"increment-button\\" type=\\"submit\\" onClick={[Function: handleClick]}>
    increment button
  </button>
</div>"
`;
